samples/event-sourcing-esdb-simple/src/main/java/io/eventdriven/ecommerce/core/scopes/InMemoryRequestAttributes.java

```diff
@@ -0,0 +1,62 @@
+package io.eventdriven.ecommerce.core.scopes;
+
+import org.springframework.web.context.request.AbstractRequestAttributes;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class InMemoryRequestAttributes extends AbstractRequestAttributes {
+
+  protected Map<String, Object> attributes = new HashMap<>();
+
+  @Override
+  public Object getAttribute(String name, int scope) {
+    return attributes.get(name);
+  }
+
+  @Override
+  public void setAttribute(String name, Object value, int scope) {
+    attributes.put(name, value);
+  }
+
+  @Override
+  public void removeAttribute(String name, int scope) {
+    attributes.remove(name);
+  }
+
+  @Override
+  public String[] getAttributeNames(int scope) {
```

Comment:
Is this `scope` parameter not used/referenced at all inside the method body?
If it isn't, I suggest to remove this `int scope` as parameter.

Reply:
Deleted this class, as it appeared that I could use `ApplicationEventPublisher` instead of hand made Event Bus with custom scoping 👍